### PR TITLE
Create Gradle cache-compliant metalava tasks.

### PR DESCRIPTION
**Pull request by vkryachko**
The original tasks were not properly cacheable and will fail validation
in future Gradle versions.

### CODE REVIEW

File: buildSrc/src/main/java/com/google/firebase/gradle/plugins/Metalava.kt [hunk 1]
```diff
@@ -14,83 +14,203 @@
...
 import org.gradle.api.tasks.TaskAction
 
 val Project.metalavaConfig: Configuration
     get() =
         configurations.findByName("metalavaArtifacts")
-                ?: configurations.create("metalavaArtifacts") {
-                    this.dependencies.add(this@metalavaConfig.dependencies.create("com.android:metalava:1.3.0"))
-                }
+            ?: configurations.create("metalavaArtifacts") {
+                this.dependencies.add(this@metalavaConfig.dependencies.create("com.android.tools.metalava:metalava:1.0.0-alpha06"))
```
Comment (rlazo) [inline buildSrc/src/main/java/com/google/firebase/gradle/plugins/Metalava.kt:37]: question: is this a newer or older version?

File: buildSrc/src/main/java/com/google/firebase/gradle/plugins/Metalava.kt [hunk 2]
```diff
@@ -14,83 +14,203 @@
...
 ) {
     val allArgs = listOf(
-            "--no-banner",
-            "--hide",
-            "HiddenSuperclass" // We allow having a hidden parent class
+        "--no-banner",
+        "--hide",
+        "HiddenSuperclass", // We allow having a hidden parent class
+        "--hide",
+        "HiddenAbstractMethod"
```
Comment (rlazo) [inline buildSrc/src/main/java/com/google/firebase/gradle/plugins/Metalava.kt:50]: why was this add?

File: firebase-database/ktx/api.txt [hunk 3]
```diff
@@ -2,13 +2,12 @@
...
 
   public final class DatabaseKt {
-    ctor public DatabaseKt();
     method @NonNull public static com.google.firebase.database.FirebaseDatabase database(@NonNull com.google.firebase.ktx.Firebase, @NonNull String url);
     method @NonNull public static com.google.firebase.database.FirebaseDatabase database(@NonNull com.google.firebase.ktx.Firebase, @NonNull com.google.firebase.FirebaseApp app);
     method @NonNull public static com.google.firebase.database.FirebaseDatabase database(@NonNull com.google.firebase.ktx.Firebase, @NonNull com.google.firebase.FirebaseApp app, @NonNull String url);
     method @NonNull public static com.google.firebase.database.FirebaseDatabase getDatabase(@NonNull com.google.firebase.ktx.Firebase);
-    method @Nullable public static inline <reified T> T getValue(@NonNull com.google.firebase.database.DataSnapshot);
-    method @Nullable public static inline <reified T> T getValue(@NonNull com.google.firebase.database.MutableData);
+    method public static inline <reified T> T getValue(@NonNull com.google.firebase.database.DataSnapshot);
```
Comment (rlazo) [inline firebase-database/ktx/api.txt:9]: This has changed from nullable to not annotated. Would this be an issue?

Here and below

File: buildSrc/src/main/java/com/google/firebase/gradle/plugins/Metalava.kt (hunk 2 above)
Comment (vkryachko) [reply buildSrc/src/main/java/com/google/firebase/gradle/plugins/Metalava.kt:50]: This is a new check metalava added, similar to the one you fixed in piper for auth

File: buildSrc/src/main/java/com/google/firebase/gradle/plugins/Metalava.kt (hunk 1 above)
Comment (vkryachko) [reply buildSrc/src/main/java/com/google/firebase/gradle/plugins/Metalava.kt:37]: this is a newer version

File: firebase-database/ktx/api.txt (hunk 3 above)
Comment (vkryachko) [reply firebase-database/ktx/api.txt:9]: I've been wondering that same thing, filed a bug internally http://b/242720906

**Review by rlazo — APPROVED** [review]

**Pull request merged**